cli/command/stream.go

```diff
@@ -0,0 +1,78 @@
+package command
+
+import (
+	"errors"
+	"github.com/Sirupsen/logrus"
+	"github.com/docker/docker/pkg/term"
+	"os"
+	"runtime"
+)
+
+// CommonStream is an input stream used by the DockerCli to read user input
+type CommonStream struct {
+	fd         uintptr
+	isTerminal bool
+	state      *term.State
+}
+
+// FD returns the file descriptor number for this stream
+func (s *CommonStream) FD() uintptr {
+	return s.fd
+}
+
+// IsTerminal returns true if this stream is connected to a terminal
+func (s *CommonStream) IsTerminal() bool {
+	return s.isTerminal
+}
+
+// SetRawTerminal sets raw mode on the input terminal
+func (s *CommonStream) SetRawTerminal() (err error) {
+	if os.Getenv("NORAW") != "" || !s.isTerminal {
+		return nil
+	}
+	s.state, err = term.SetRawTerminal(s.fd)
+	return err
+}
+
+// RestoreTerminal restores normal mode to the terminal
+func (s *CommonStream) RestoreTerminal() {
+	if s.state != nil {
+		term.RestoreTerminal(s.fd, s.state)
+	}
+}
+
+// GetTtySize returns the height and width in characters of the tty
+func (s *CommonStream) GetTtySize() (uint, uint) {
+	if !s.isTerminal {
+		return 0, 0
+	}
+	ws, err := term.GetWinsize(s.fd)
+	if err != nil {
+		logrus.Debugf("Error getting size: %s", err)
+		if ws == nil {
+			return 0, 0
+		}
+	}
+	return uint(ws.Height), uint(ws.Width)
+}
+
+// CheckTty checks if we are trying to attach to a container tty
+// from a non-tty client input stream, and if so, returns an error.
+func (s *CommonStream) CheckTty(attachStdin, ttyMode bool) error {
```

Comment:
This only existed on `InStream`, right?  

It doesn't seem to apply to `OutStream`, could this be moved back to `InStream` ?